InvenTree/part/serializers.py

```diff
@@ -592,7 +592,7 @@ def __init__(self, *args, **kwargs):
 
         if not create:
             # These fields are only used for the LIST API endpoint
-            for f in self.skip_create_fields()[1:]:
+            for f in self.skip_create_fields()[2:]:
```

Comment:
Instead of using the index offset here (`[2:]`) I would suggest doing a string comparison:

```python
for f in self.skip_create_fields():
    if f == 'existing_image':
        # existing_image field is also support for PATCH requests, but does not exist on the model
        continue
    self.fields.pop(f)
```

This way we do not have to enforce list ordering, and it is clear *why* we are doing this custom operation

Reply:
Looking at the indexing issue, the original index of `[:1]` was put there to assure that the `remote_image` field of the `RemoteImageMixin` isn't removed either. It's due to this that I proposed a potential whitelist concept.

Then, you could formulate the loop like this:

```python
for f in self.skip_create_fields():
    if f in ["remote_image", "existing_image"]:
        continue
    self.fields.pop(f)
```

Reply:
That makes sense. Much cleaner and obvious outcome!